### DISCUSSION TITLE
Wrapping Budibase App into a mobile app?

### DISCUSSION THREAD

**Discussion by supermx1** [Budibase/budibase#1084] [Ideas]
An idea for the project will be the ability to wrap a budibase project in a PWA or Mobile App (Native or otherwise). Many app users use their mobile phones for daily transactions. So having a way to port a budibase project to mobile will be super amazing. Using Capacitor or PhoneGap or other ideas out there.

**Comment by mjashanks** [comment]
Agreed! In time. Allowing responsive design is the big piece of work here. Definitely on the radar.

**Reply by Kuuchuu** [reply]
Hello @mjashanks , I just wanted to see if there have been any updates to allow for easy deployment as a PWA, or if you know of any ways anyone has been successful in creating a PWA from their Budibase site. Thanks for all you do!

**Reply by mjashanks** [reply]
Hi @Kuuchuu it's not possible right now, but we are hoping to get this done soon - potentially in March 2022.

**Comment by v3ss0n** [comment]
Anyprogress on PWA?

**Reply by validlogin** [reply]
Interesting concept!

**Reply by siriani** [reply]
Waiting for this

**Reply by dewdeded** [reply]
Why? I mean if u really need PWA, you can wrap a free PWA solution like Capacitor or Cordova around the Budibase app.
What is the use case?
Budibase is a business oriented plattform, so most mobile device users will be inside an MDM solution.

**Comment by AlexisLeza** [comment]
Hi;
Any update about PWA?

**Comment by jibe-b** [comment]
+1

**Comment by dewdeded** [comment]
We deploy our regular Budibase apps via KACE Mobile Device Management to the mobile devices (mobile phones and tables) of our employees. (iOS and Android). KACE MDM pushes the app we want to the correct devices and the phones have a "kiosk mode/view" for webapps, with native icon, etc. (All our users have VPN on their phones.) Budibase apps look like native apps on the phones.

For some very important managers and few selected BB apps we have created an Azure App Registry for the specific app.
IT requires Azure Active Directory Auth before reverse-proxying the request into our corporate DMZ where the Budibase instance is. 
So these managers (< 5) can use Budibase over the internet, while not being in our corporate LAN (or at least VPN) or when using unmanaged clients (private hardware, hardware from other companies they work for) or if they are connected to a VPN blocking network like in hotels, ...). But the reverse-proxing via Azure helps us, to enforce strict access rules & monitoring, only very few approved guys are allowed to do so and we don't have to expose Budibase to the public internet.
